alexandria/searching/src/binary_search.cairo

```diff
@@ -0,0 +1,34 @@
+use array::SpanTrait;
+
+fn binary_search<T,
+impl TCopy: Copy<T>,
+impl TDrop: Drop<T>,
+impl TEq: PartialEq<T>,
+impl TOr: PartialOrd<T>>(
+    span: Span<T>, val: T
```

Comment:
Shouldn't this be a snapshot ?

Reply:
I could try to change it but all the Span methods expect a Span and not a snapshot